common/src/main/java/io/netty/util/internal/BlockingMessageQueue.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2022 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util.internal;
+
+import org.jctools.queues.MessagePassingQueue;
+
+import java.util.ArrayDeque;
+
+/**
+ * This is an implementation of {@link MessagePassingQueue}, similar to what might be returned from
+ * {@link PlatformDependent#newMpscQueue(int)}, but intended to be used for debugging purpose.
+ * The implementation relies on synchronised monitor locks for thread-safety.
+ * The {@code drain} and {@code fill} bulk operations are not supported by this implementation.
+ */
+@UnstableApi
+public final class BlockingMessageQueue<T> implements MessagePassingQueue<T> {
+    private final ArrayDeque<T> deque;
```

Comment:
Can you add some comments why we did use an `ArrayDeque` and not a `ConcurrentLinkedQueue` ?